### ISSUE TITLE
Headend only returns Azure preview offer/plan info

### ISSUE THREAD

**Issue by jefflill** [nforgeio/neonKUBE#1843]
While looking into automating publication of our Azure node images, I noticed that the headend is always returning the marketplace information for the **preview** release.

This worked for us and Loopie because we explicitly added the Loopie and Marcus personal subscriptions to the preview security list.  This presumably wouldn't have worked for anybody else because their subscription wouldn't be on the list.

- [x] Modify the Azure endpoint on the Headend to accept a boolean **preview** parameter.  When this is TRUE, the headend will return details for the preview image otherwise it will return the public details.  I believe I can make this parameter optional, so we won't need a new API version.
- [x] Add a **--use-preview** option to the **neon cluster prepare** and **deploy** commands
    - [x] kubectl
    - [x] neon-cli
- [x] Modify the Azure hosting manager's **LocateNodeImageAsync()** method to query get the new **preview** flag from somewhere and pass this to the modifed headend API.

**Comment by jefflill** [comment]
**BUG:** I just noticed that the **neon cluster prepare ...** command incorrectly accepts the **--stage** option instead of **--use-staged[=BRANCH]** and the comment for this is also incorrect.

We need to rename this and update:

- [ ] kubectl
- [ ] neon-cli
- [ ] cheats
    - [ ] Jeff
    - [ ] Marcus

**Comment by jefflill** [comment]
So, I thought that we needed to add a **-preview** suffix to the offer/product name to install the preview, but this isn't working, even after I authorized the neonforge subscription ID.  I'm guessing now that we don't need to add this suffix and the preview image will be automatically used if it's the only one present and the subscription is authorized.  I've done some googling, and haven't seen anything that conflicts with that idea.

I'm going to re move the `--use-preview` option and any related code and then close this issue.  I'll try to remember to test this after we've published the next release preview.
